libs/SalesforceSDKCore/SalesforceSDKCore/Classes/NativeLoginManager.swift

```diff
@@ -44,7 +44,7 @@ public protocol NativeLoginManager {
     ///   - username: User provided Salesforce username.
     ///   - password: User provided Salesforce password.
     /// - Returns: NativeLoginResult
-    @objc func login(username: String, password: String) async -> NativeLoginResult
+    @objc func login(username: String, password: String) async throws -> NativeLoginResult
```

Comment:
Personally, I don't think login should throw.  Maybe there is a case to be made for a purely dev setup scenario like `reCaptchaSiteKeyId` is not provided and they try to imitate OTP.  But having the ability for the app to crash on the user because the dev doesn't catch it feels wrong.  IMO a string message about something that went wrong is just as useful in a log as it is in a throwable at runtime.

Reply:
I could roll with just logging a warning and returning one of the int enum types as well.

Reply:
I'm trying out the suggestion by @brandonpage to see what it would look like if the native login manager methods catch all error and unexpected conditions, simply log them and return one of the NativeLoginResult int codes.  If that looks good, that'll be our way forward on this one.

Reply:
I don't think we have a specific style set, but looking through the other swift code, we have throwing methods in [Encryptor.swift](https://github.com/forcedotcom/SalesforceMobileSDK-iOS/blob/dev/libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/Encryptor.swift) and Swift only result methods in extensions like [SmartStore.swift](https://github.com/forcedotcom/SalesforceMobileSDK-iOS/blob/dev/libs/SmartStore/SmartStore/Classes/Extensions/SmartStore.swift). I could see any of those (with an objective-c/swift combo if needed) or the original signature as long as all the errors are in one place (like if throwing, move over the errors from `NativeLoginResult`)